home-excercises-05-2022/src/main/java/io/codelex/secondT/exercise2/Basket.java

```diff
@@ -0,0 +1,44 @@
+package io.codelex.secondT.exercise2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket<E> {
+
+    private int count = 0;
+    List<E> contents;
+
+    public Basket() {
+        contents = new ArrayList<>();
+    }
+
+    public void addToBasket(E element) {
+        if (count == 10) {
+            throw new BasketFullException("Basket is full :(");
+        } else {
+            contents.add(element);
+            count++;
+        }
+    }
+
+    public void removeFromBasket() {
+        if (count == 0) {
+            throw new BasketEmptyException("Basket is empty :(");
+        } else {
+            contents.remove(contents.size() - 1);
+            count--;
+        }
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+    public void setCount(int count) {
```

Comment:
Īsti laikam nebūtu korekti ļaut uzstādīt to count, jo, ja kāds to pasauc, tad count var nesakrist ar to kas faktiski ir iekš list.

Reply:
setCount() metodi izveidoju tikai, lai ātri testā testFullBasketException() var Basket pilnu ātri izveidot, bet tagad šo metodi izdzēsu un testu arī pārveidoju bez šīs metodes